NETDefault/WebApp/Controllers/CacheDependentController.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+
+namespace WebApp.Controllers
+{
+    /// <summary>
+    /// see Health check and application warmup
+    /// </summary>
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CacheDependentController : ControllerBase
+    {
+        private readonly ILogger<CacheDependentController> _logger;
+        private readonly IMemoryCache _memoryCache;
+
+        public CacheDependentController(
+            ILogger<CacheDependentController> logger,
+            IMemoryCache memoryCache)
+        {
+            _logger = logger;
+            _memoryCache = memoryCache;
+        }
+
+        [HttpGet]
+        public ActionResult<string> GetString()
+        {
+            var value = _memoryCache.Get<string>("CacheItem");
+
+            if (!string.IsNullOrWhiteSpace(value))
+            {
+                _logger.LogInformation("Cache: ok");
+                return $"{value}";
+            }
+
+            _logger.LogInformation("Cache: not-ok");
+            return NotFound();
+        }
+
+        [HttpGet("check")]
```

Comment:
If I would like to have different types of cache keys, I might wonder how to format the URL.

_:cacheKey/check_?

Reply:
I think that should be based on the specific requirements of the project that is needing such controller. I don't want to include such features in this example project as that quickly becomes a copy-pasta that is not always needed or appropriate. I think these example projects should demonstrate a concept and not provide a base-line implementation.